src/NumericalAlgorithms/SphericalHarmonics/YlmSpherepack.cpp

```diff
@@ -732,7 +678,48 @@ T YlmSpherepack::interpolate_from_coefs(
   return result;
 }
 
-void YlmSpherepack::calculate_interpolation_data() const {
+void YlmSpherepack::calculate_collocation_points() {
+  // Theta
+  auto& theta = storage_.theta;
+  DataVector temp(2 * n_theta_ + 1);
```

Comment:
`DataVector` is better than `std::vector<double>`. It default initializes to `std::signaling_NaN()` in debug (helps catch errors) and doesn't default initialize in release (more efficient). In general, I think we should use fewer `std::vector<double>` and more `DataVector`. Note that Blaze's vectors would not be a suitable replacement for `DataVector` since we have no guarantees on how they initialize memory.

Reply:
Ok. Tangent unrelated to this PR: my concern with overusing DataVector is that it has NodalVector semantics in many places, meaning that it represents data at grid points and supports associated arithmetics, as opposed to ModalVector that explicitly forbids some operations. Perhaps we should keep DataVector as a general-purpose dense vector with the memory allocation behavior that you mentioned, and add a new NodalVector type specifically for data at grid points. What do you think?